Automattic/woocommerce-payments
# refactor: slight refactor on the UPE available methods

Fixes #

#### Changes proposed in this Pull Request









Ensures that when the UPE flag is disabled, only `card` is returned as an available payment method.

Some inconsistency has been created.
We needed to add additional available payment methods. But UPE was not enabled.
So we created the `get_upe_available_payment_methods` method with the `wcpay_upe_available_payment_methods` filter.
Then the UPE gateway got introduced, and inherited the method. All good.
But as work progressed, we forgot to _only_ return the additional payment methods when UPE is enabled.

Adding some consistency, so that there's less confusion if the flags are manually switched.

#### Testing instructions





- With "grouped settings" enabled
  - Just UI changes on the settings
- With "grouped settings" and "Enable UPE settings changes"
  - Adds "Payments accepted on checkout" section in the new settings: https://d.pr/i/TZKnu5
- With "grouped settings" and "Enable UPE checkout"
  - Just UI changes on the settings
  - Adds "Setup additional payment methods" task: https://d.pr/i/mt7n4b
- With "grouped settings" and "Add UPE additional payment methods"
  - Just UI changes on the settings - this is not a valid combination of flags we're going to release ("Add UPE additional payment methods" is technically dependent on "Enable UPE checkout")


Feel free to toggle between feature flags, things should be more consistent, now.

Having "grouped settings" and "Enable UPE checkout" is the only "fishy" one, because it adds the "Setup additional payment methods" task.
But we're going to launch with that at the moment, so I personally think it makes sense.

-------------------
